src/icalendar_searcher/__init__.py

```diff
@@ -0,0 +1,212 @@
+    which means completed tasks will be tfiltered out.
+
+    ``start`` and ``end`` is giving a time range.  The CalDAV logic
+    will be honored, see RFC4791, section 9.9 for very clear
+    definitions of what should be returned and what should be filtered
+    away.  Timestamps should ideally be with a time zone, if not given
+    the local time zone will be assumed.  All-day events may be tricky
+    to get correct when timestamps are given and calendar data covers
+    multiple time zones.
+
+    ``alarm_start`` and ``alarm_end`` is similar for alarm searching
+
+    If ``expand`` is set to True, recurring objects will be expanded
+    into reccurence objects for the time period specified by ``start``
+    and ``end``.
+
+    Unless you now what you are doing, ``expand`` should probably be
+    set to true - but since ``expand`` can only be combined with a
+    start and end, it's by default set to false.
+
+    For filtering an icalendar instance ``mycal`` containing a
+    VCALENDAR with multiple independent events, one may do
+    ``searcher.filter([mycal])``.  The CalDAV library contains a
+    CalDAVSearcher class inheritating this class and including a
+    ``.search(caldav_calendar)`` method.  The idea is that it may be
+    done in the same way also for searching other calendars or
+    calendar-like systems using other protocols.
+
+    The filtering and sorting methods should accept both wrapper
+    objects (for the CalDAV library, those are called
+    CalendarObjectResource and contains extra information like the URL
+    and the client object) and ``icalendar.Calendar`` objects from the
+    icalendar library.  Wrapper objects should have the
+    ``icalendar.Calendar`` object available through a property
+    ``icalendar_instance``.  Even for methods expecting a simple
+    component, a ``Calendar`` should be given.  This is of
+    imporantance for recurrences with exceptions (like, "daily meeting
+    is held at 10:00, except today the meeting is postponed until
+    12:00" may be represented through a calendar with two
+    subcomponents)
+
+    Methods expecting multiple components will always expect a list of
+    calendars (CalDAV-style) - but the algorithms should also be
+    robust enough to handle multiple independent components embedded
+    in a single Calendar.
+
+    Other ideas that one may consider implementing:
+    * limit, offset.
+    * fuzzy matching
+
+    """
+
+    todo: bool = None
+    event: bool = None
+    journal: bool = None
+    start: datetime = None
+    end: datetime = None
+    alarm_start: datetime = None
+    alarm_end: datetime = None
+    comp_class: Union["CalendarObjectResource", "Component"] = None
+    include_completed: bool = None
+
+    expand: bool = False
+
+    _sort_keys: list = field(default_factory=list)
+    _property_filters: dict = field(default_factory=dict)
+    _property_operator: dict = field(default_factory=dict)
+
```

Comment:
```suggestion
    def __post_init__(self):
        """Set include_completed default based on todo flag."""
        if self.include_completed is None:
            self.include_completed = not self.todo if self.todo is not None else True

    def add_property_filter(
```

Docstring (lines 52-53) says `include_completed` defaults to `False` when `todo=True`, but no `__post_init__` implements this.